### ISSUE TITLE
Bug: React roots are announced as clickable to screen readers

### ISSUE THREAD

**Issue by timogasda** [facebook/react#20895]
React version: 17.0.1

## Steps To Reproduce

1. Activate your screen reader
   1. I used [NVDA](https://www.nvaccess.org/), but [JAWS](https://www.freedomscientific.com/products/software/jaws/) would also work
   1. This will not work with VoiceOver because it does not have the same heuristic about elements with click handlers
2. Go to a page that has one or more React 17 root nodes. For example: https://qc9xj.csb.app/
3. Use the regular navigation of your screen reader to explore the page
   1. In the case of NVDA, this would be using Arrow keys up and down
4. Hear your screen reader announcing the H2 as "clickable"

Link to code example: https://codesandbox.io/s/epic-babbage-qc9xj (https://qc9xj.csb.app/)

## The current behavior

Here is the log of what NVDA says as I'm navigating the page:

> content info landmark    heading    level 1  This is a static HTML footer
>     _(I press Arrow Down)_
> clickable  heading    level 2  This is a heading within a React application.
>     _(I press Arrow Down)_
> banner landmark    heading    level 1  This is a static HTML header

## The expected behavior

It should not announce the heading as clickable/interactive. Here is what it would sound like without React on the page:

> content info landmark    heading    level 1  This is a static HTML footer
>     _(I press Arrow Down)_
> heading    level 2  This is a heading within a React application.
>     _(I press Arrow Down)_
> banner landmark    heading    level 1  This is a static HTML header

It should be noted that this is how it used to work in React 16 because the delegation of events was done on the `document` node, not the React root node.

## Notes
* Some screen readers like JAWS and NVDA use the presence of a `click` handler on a node as an indicator that they are interactive
* Because of React's event delegation, all synthetic events are handled at one central node. In React 16 it's the `document` and in React 17 it's the root node.
* The example shows that it's announced as clickable even when there is no interactive content inside. But even if the React node contained interactive content, the parent container should not be announced as interactive - because it is not.
* This is also called out by the Accessibility Inspector in the Firefox DevTools:
  * "Does not meet WCAG standards for accessibility. Clickable elements must be focusable and should have interactive semantics. [Learn more](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics)"
* While this is already an issue with just one React root on the page, this problem will get increasingly worse and annoying for the user when there are multiple React roots on the page
    * This could easily happen on pages that are constructed with micro frontend frameworks or widgets

**Comment by eps1lon** [comment]
Thanks for the report.

We've had similar issues with this in Material-UI. 

Following their rationale, any element in a given DOM subtree is clickable. So as soon as you have a click handler on the document, every element is clickable.

Though I understand that screen readers want to patch up potential author errors, they should ultimately add a way to ensure that these elements are in fact not actually clickable. In other words, let authors declare that a certain event listener is used for event delegation.

I would encourage you file issues for each respective assistive technologies on channels they own (e.g. https://github.com/nvaccess/nvda/issues) and ask how authors should declare that a listener is used for event delegation.

PS: A similar issue can occur if you implement "click away listeners". They could check if a click is not contained in a certain region. This would lead to that region being announced as "clickable" even though   we don't want to communicate that you should click this region.

**Comment by NullPainter2** [comment]
@eps1lon If I do not specify `onClick` handler in JSX, but React adds event handler like `noop()`, then it is a bug in React. And not in NVDA.

**Comment by eps1lon** [comment]
> @eps1lon If I do not specify `onClick` handler in JSX, but React adds event handler like `noop()`, then it is a bug in React. And not in NVDA.

Even if React wouldn't add a noop in your specific case, it would still add actual onclick handlers in other cases. So while React might be able to fix that specific case, NVDA is in a way better position to not consider this an author error. Authors should be able to explicitly tell NVDA that an element is clickable. Just because an element has a click handler, does not mean clicking that element triggers any action or that the click handlers has relevance to that element (e.g. click-away listeners).

**Comment by NullPainter2** [comment]
@eps1lon Well, my problem is not the other cases ;) .

I have React 16.x (but 17.x has same error as 16.x). 

I have different problem than OP here. 

My problem is not with **first** element inside of react root which NVDA reports as "clickable" (which OP has), but with element which is deep down into the React tree. Also when I do not specify `onClick()`, then NVDA does not read element as "clickable". NVDA reads element as "clickable" when React adds `noop()` **onto** the element (which happens after resize ...).

Also from MDN article which OP linked I quote _"An element is clickable if it has an onclick event handler defined."_ . This article from Mozilla follows W3C standard for web accessibility, so React does not follow web standard here.

Please reopen my bug.

**Comment by eps1lon** [comment]
As @feerrenrut pointed out (see ), we can fix this in React applications by adding `role="presesenation"` to React root containers e.g. https://codesandbox.io/s/dont-announce-react-root-children-as-clickable-y8xfy?file=/public/index.html.

Will do some more testing if we can improve the current situation either by adding a warning or automatically fixing it. One thing that needs to be checked is how Portals affect this issue.

**Comment by jsrobson10** [comment]
one workaround i have found works (or at least in firefox accessibility tools) is to set all root elements to aria-disabled="true". this will just tell screen readers that while the root "button" is "clickable", it is disabled. setting role="presentation" or role="none" is not an option as it hides the root nodes children.

**Comment by a11y1** [comment]
I want to reopen this issue. Experiencing this in my applications.
